pkg/shell/base_index.js

```diff
@@ -356,8 +367,19 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
             if (typeof data !== "string")
                 return;
 
-            var source = source_by_name[child.name];
-            var control;
+            var source, control;
+
+            /*
+             * On Internet Explorer we see Access Denied when non Cockpit
+             * frames send messages (such as Javascript console). This also
+             * happens when the window is closed.
+             */
+            try {
+                source = source_by_name[child.name];
+            } catch(ex) {
+                return;
```

Comment:
Blanket catches always smell dubious, as they are prone to silently ignoring real/unexpected exceptions. How does such an `ex` for an Access Denied look like? Is there anything predictable in it which this `catch` could use to restrict this? Could this at least `console.warn(ex)` for debugging?